go/toolchains.rst

```diff
@@ -18,22 +27,568 @@ customize the behavior of the core_ go rules.
+
+compile
+~~~~~~~
+
+The compile function adds an action that runs ``go tool compile`` on a set of source files
+to produce an archive.
+
+It does not return anything.
+
++--------------------------------+-----------------------------+-----------------------------------+
+| **Name**                       | **Type**                    | **Default value**                 |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`ctx`                   | :type:`string`              | |mandatory|                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| The current rule context, used to generate the actions.                                          |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`go_toolchain`          | :type:`the go toolchain`    | |mandatory|                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| This must be the same go toolchain object you got this function from.                            |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`sources`               | :type:`list<File>`          | |mandatory|                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| An iterable of source code artifacts.                                                            |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`importpath`            | :type:`string`              | :value:`""`                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| The import path this package represents. This is passed to the -p flag.                          |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`golibs`                | :type:`list<GoLibrary>`     | :value:`[]`                       |
++--------------------------------+-----------------------------+-----------------------------------+
+| An iterable of all imported libraries.                                                           |
++--------------------------------+-----------------------------+-----------------------------------+
+| :param:`mode`                  | :type:`string`              | :value:`NORMAL_MODE`              |
++--------------------------------+-----------------------------+-----------------------------------+
+| Controls the compilation setup affecting things like enabling profilers and sanitizers.          |
+| This must be one of the values in common.bzl#compile_modes.                                      |
```

Comment:
Would be better to list the valid modes with short descriptions here. I know it's redundant, but common.bzl is private and may change, and it's not immediately clear what's allowed from modes.rst.